pyhx870/protocol.py

```diff
@@ -183,6 +183,9 @@ def wait_for_ready(self, timeout=1):
         if radio_status != "00":
             raise TimeoutError("Radio not ready")
 
+    def set_progress_callback(self, callback):
+        self.progress_callback = callback
+
```

Comment:
I like the idea of a callback, and it's difinitely the way to go for more complex progress logging setups, but for this I prefer a way simple approach.